openshift-examples/keycloak.yaml

```diff
@@ -60,40 +60,40 @@ objects:
         spec:
           containers:
             - env:
-                - name: KEYCLOAK_USER
-                  value: '${KEYCLOAK_USER}'
-                - name: KEYCLOAK_PASSWORD
-                  value: '${KEYCLOAK_PASSWORD}'
-                - name: DB_VENDOR
-                  value: '${DB_VENDOR}'
-                - name: JGROUPS_DISCOVERY_PROTOCOL
-                  value: dns.DNS_PING
-                - name: JGROUPS_DISCOVERY_PROPERTIES
-                  value: 'dns_query=${APPLICATION_NAME}.${NAMESPACE}.svc.cluster.local'
+                - name: KEYCLOAK_ADMIN
+                  value: '${KEYCLOAK_ADMIN}'
+                - name: KEYCLOAK_ADMIN_PASSWORD
+                  value: '${KEYCLOAK_ADMIN_PASSWORD}'
+                - name: KC_PROXY
+                  value: 'edge'
               image: quay.io/keycloak/keycloak:17.0.0
               livenessProbe:
-                failureThreshold: 3
+                failureThreshold: 100
                 httpGet:
-                  path: /auth/realms/master
+                  path: /
                   port: 8080
                   scheme: HTTP
                 initialDelaySeconds: 60
               name: '${APPLICATION_NAME}'
               ports:
                 - containerPort: 8080
                   protocol: TCP
-                - containerPort: 8443
-                  name: https
-                  protocol: TCP
               readinessProbe:
-                failureThreshold: 10
+                failureThreshold: 300
                 httpGet:
-                  path: /auth/realms/master
+                  path: /
                   port: 8080
                   scheme: HTTP
                 initialDelaySeconds: 30
               securityContext:
                 privileged: false
+              volumeMounts:
+                - mountPath: /opt/keycloak/data
+                  name: empty
+              args: ["start-dev"]
+          volumes:
+            - name: empty
+              emptyDir: {}
```

Comment:
jFTR, created #307.

Reply:
@andreaTP With https://github.com/keycloak/keycloak/pull/10783 merged, I think we can remove the volume mount. WDYT?

Reply:
@vmuzikar we need a release before removing this.

Reply:
This is `main` branch, not `latest`. That means this PR is already targeted for the next release. 🙂

Reply:
The image used in `keycloak.yaml` is `17.0.0`, we can possibly remove the workaround when we update the image version.

Reply:
Again, `main` branch is the dev branch. 🙂 It's a preparation for 18.0.0 regardless to what's currently in the `image` field. So it should be ok to change it now. Is that correct @stianst?

Reply:
I think this is a not important detail:
 - mounting an external volume is actually good practice
 - when we switch to the next version in the image field we can remove the workaround (otherwise we are going to have a broken template again for the time being)
 
 Overall, whatever works for you.

Reply:
Agree it's not a very important detail. It just feels a bit wrong to be introducing a "workaround" for a problem of the past. 😉

It is ok to break the template in `main` branch as it is a "future" branch for the next release. That's a standard process. Users are supposed to use `latest` branch.

But ok, I don't want to block this PR, so approving.